### PR TITLE
[docs] [voyager] added arguments and commands to yb-voyager-cli for 0.81

### PR DESCRIPTION
**Pull request by aishwarya24**
Build-preview : https://deploy-preview-14605--infallible-bardeen-164bc9.netlify.app/preview/migrate/yb-voyager-cli/

### CODE REVIEW

File: docs/content/preview/migrate/yb-voyager-cli.md [hunk 1]
```diff
@@ -265,12 +266,28 @@ Example : `filename1:tablename1,filename2:tablename2[,...]`
...
+
+### --file-opts
+
+Comma-separated string options for CSV file format. The options can include the following:
+
+1. `escape_char`: escape character
+
+1. `quote_char`: character used to quote the values
+
+Default: double quotes '"'
```
Comment (ddhodge) [inline docs/content/preview/migrate/yb-voyager-cli.md:285]: is this working the way you intend?
is there a default escape character?

File: docs/content/preview/migrate/yb-voyager-cli.md [hunk 2]
```diff
@@ -265,12 +266,28 @@ Example : `filename1:tablename1,filename2:tablename2[,...]`
...
 
 Default: false; change to true if the CSV file contains column names as a header.
 
+**Note**: Boolean flags takes arguments in the format `--flag-name=[true|false]`, and not `--flag-name [true|false]`.
+
+### --file-opts
+
+Comma-separated string options for CSV file format. The options can include the following:
+
+1. `escape_char`: escape character
```
Comment (ddhodge) [inline docs/content/preview/migrate/yb-voyager-cli.md:281]: do they need to be in a specific order?

**Review by ddhodge — COMMENTED** [review]
couple questions

File: docs/content/preview/migrate/yb-voyager-cli.md (hunk 2 above)
Comment (sanyamsinghal) [reply docs/content/preview/migrate/yb-voyager-cli.md:281]: Any random order would work. Internally identifies and use it accordingly

Comment (ddhodge) [reply docs/content/preview/migrate/yb-voyager-cli.md:281]: Use bullets then

File: docs/content/preview/migrate/yb-voyager-cli.md (hunk 1 above)
Comment (sanyamsinghal) [reply docs/content/preview/migrate/yb-voyager-cli.md:285]: thanks for pointing it out @ddhodge
1. Yes, there is a default escape character, it is also double quotes.
2. Also, We should mention somewhere that escape_char and quote_char are only valid/required for CSV file format(which is the default format)

Comment (aishwarya24) [reply docs/content/preview/migrate/yb-voyager-cli.md:285]: @sanyamsinghal , what are the other file formats that we need to mention apart from CSV. Are others supported?

Comment (sanyamsinghal) [reply docs/content/preview/migrate/yb-voyager-cli.md]: Supported formated are: csv and text

File: docs/content/preview/migrate/yb-voyager-cli.md [hunk 3]
```diff
@@ -265,12 +266,30 @@ Example : `filename1:tablename1,filename2:tablename2[,...]`
...
+
+### --file-opts
+
+Comma-separated string options for CSV file format. The options can include the following:
+
+- `escape_char`: escape character
+
+- `quote_char`: character used to quote the values
+
+Default (for both options): double quotes '"'
```
Comment (ddhodge) [inline docs/content/preview/migrate/yb-voyager-cli.md:285]:
```suggestion
Default: double quotes (") for both escape and quote characters
```

File: docs/content/preview/migrate/yb-voyager-cli.md [hunk 4]
```diff
@@ -265,12 +266,30 @@ Example : `filename1:tablename1,filename2:tablename2[,...]`
...
+
+Note that `escape_char` and `quote_char` are only valid/required for CSV file format.
```
Comment (ddhodge) [inline docs/content/preview/migrate/yb-voyager-cli.md:287]:
```suggestion
Note that `escape_char` and `quote_char` are only valid and required for CSV file format.
```

**Review by ddhodge — APPROVED** [review]

File: docs/content/stable/migrate/yb-voyager-cli.md [hunk 5]
```diff
@@ -269,6 +312,57 @@ This argument is to be specified only for CSV file type.
...
+Default : true
+
+### --yes
+
+By default, answer yes to all questions during migration.
+
+### --start-clean
+
+Clean the project's data directory for already existing files before starting migration.
+
```
Comment (rahulb-yb) [inline docs/content/stable/migrate/yb-voyager-cli.md:363]: Please also add the documentation for `--table-list` and `--exclude-table-list` in this section, I believe it has been missed in this iteration of the PR. Picked up straight from the [google doc](https://docs.google.com/document/d/1dZ7RTzKAhMgs5W9OR4NvwjtZW2M7fh3QMGpfBvQCYYM/edit#):
```
--table-list (string)             Comma separated list of the tables to for which data is exported
--exclude-table-list (string)     Comma separated list of tables to exclude while exporting data(no-op if --table-list is used)
```

File: docs/content/stable/migrate/yb-voyager-cli.md [hunk 6]
```diff
@@ -181,10 +199,31 @@ yb-voyager import data file --export-dir /path/to/yb/export/dir \
         --target-db-user username \
         --target-db-password password \
         --target-db-name dbname \
-        –-data-dir "/path/to/files/dir/" \
+        --target-db-schema #Applicable only for MySQL and Oracle
```
Comment (rahulb-yb) [inline docs/content/stable/migrate/yb-voyager-cli.md:202]: I believe `--target-db-schema` has an argument to be passed:
```
--target-db-schema schemaname
```

File: docs/content/stable/migrate/yb-voyager-cli.md [hunk 7]
```diff
@@ -156,6 +173,7 @@ yb-voyager import data --export-dir /path/to/yb/export/dir \
         --target-db-user username \
         --target-db-password password \
         --target-db-name dbname \
+        --target-db-schema #Applicable only for MySQL and Oracle
```
Comment (rahulb-yb) [inline docs/content/stable/migrate/yb-voyager-cli.md:176]: Same as line 202

File: docs/content/stable/migrate/yb-voyager-cli.md [hunk 8]
```diff
@@ -133,7 +149,8 @@ yb-voyager import schema --export-dir /path/to/yb/export/dir \
         --target-db-host hostname \
         --target-db-user username \
         --target-db-password password \
-        --target-db-name dbname
+        --target-db-name dbname \
+        --target-db-schema #Applicable only for MySQL and Oracle
```
Comment (rahulb-yb) [inline docs/content/stable/migrate/yb-voyager-cli.md:153]: Same as line 202

File: docs/content/preview/migrate/yb-voyager-cli.md [hunk 9]
```diff
@@ -113,7 +107,29 @@ yb-voyager export data --export-dir /path/to/yb/export/dir \
...
+
+- *arguments*: See [Arguments](#arguments)
+
+#### Example
+
+```sh
+yb-voyager export data status --export-dir /path/to/yb/export/dir
 ```
 
 ### import schema
```
Comment (rahulb-yb) [inline docs/content/preview/migrate/yb-voyager-cli.md:135]: The new description of import schema should read:

During a typical migration workflow, the user is required to run the `import schema` command twice: first **without** the `post-import-data` flag and second time **with** the flag.
The second invocation **must** be done **after** the **data** import is already complete. The second invocation creates indexes and triggers in the target schema.

**Review by rahulb-yb — COMMENTED** [review]
Left a few changes to be made, synced with [google doc](https://docs.google.com/document/d/1dZ7RTzKAhMgs5W9OR4NvwjtZW2M7fh3QMGpfBvQCYYM/edit#) for future reference.

File: docs/content/preview/migrate/yb-voyager-cli.md [hunk 10]
```diff
@@ -265,12 +309,63 @@ Example : `filename1:tablename1,filename2:tablename2[,...]`
...
 Default: '\t' (tab); can be changed to comma(,), pipe(|) or any other character.
 
-### –-has-header
+### --has-header
 
 This argument is to be specified only for CSV file type.
 
 Default: false; change to true if the CSV file contains column names as a header.
 
+**Note**: Boolean flags takes arguments in the format `--flag-name=[true|false]`, and not `--flag-name [true|false]`.
```
Comment (ddhodge) [inline docs/content/preview/migrate/yb-voyager-cli.md:318]:
```suggestion
**Note**: Boolean flags take arguments in the format `--flag-name=[true|false]`, and not `--flag-name [true|false]`.
```

File: docs/content/preview/migrate/yb-voyager-cli.md [hunk 11]
```diff
@@ -265,12 +309,63 @@ Example : `filename1:tablename1,filename2:tablename2[,...]`
...
+
+### --file-opts
+
+Comma-separated string options for CSV file format. The options can include the following:
+
+- `escape_char`: escape character
+
+- `quote_char`: character used to quote the values
+
+Default : double quotes (") for both escape and quote characters
```
Comment (ddhodge) [inline docs/content/preview/migrate/yb-voyager-cli.md:328]:
```suggestion
Default: double quotes (") for both escape and quote characters
```

File: docs/content/preview/migrate/yb-voyager-cli.md [hunk 12]
```diff
@@ -265,12 +309,63 @@ Example : `filename1:tablename1,filename2:tablename2[,...]`
...
+
+### --oracle-home
+
+Path to set `$ORACLE_HOME` environment variable. `tnsnames.ora` is found in `$ORACLE_HOME/network/admin`.
+
+### --use-orafce
+
+Enable using orafce extension in export schema.
+
+Default : true
```
Comment (ddhodge) [inline docs/content/preview/migrate/yb-voyager-cli.md:357]:
```suggestion
Default: true
```

File: docs/content/preview/migrate/yb-voyager-cli.md [hunk 13]
```diff
@@ -265,12 +309,63 @@ Example : `filename1:tablename1,filename2:tablename2[,...]`
...
+
+Note that `escape_char` and `quote_char` are only valid and required for CSV file format.
+
+Example: `--file-opts "escape_char=\",quote_char=\""` or `--file-opts 'escape_char=",quote_char="'`
+
+### --format
+
+Specifies the format of your data file with CSV or text as the supported formats.
+
+Default : CSV
```
Comment (ddhodge) [inline docs/content/preview/migrate/yb-voyager-cli.md:338]:
```suggestion
Default: CSV
```

File: docs/content/preview/migrate/yb-voyager-cli.md [hunk 14]
```diff
@@ -249,7 +293,7 @@ Specifies the number of records that the [export directory](../install-yb-voyage
 
 Default : 100,000
```
Comment (ddhodge) [inline docs/content/preview/migrate/yb-voyager-cli.md:294]:
```suggestion
Default: 100,000
```

**Review by ddhodge — APPROVED** [review]
Minor comments; apply in stable as well

File: docs/content/preview/migrate/yb-voyager-cli.md [hunk 15]
```diff
@@ -265,12 +309,63 @@ Example : `filename1:tablename1,filename2:tablename2[,...]`
...
+### --format
+
+Specifies the format of your data file with CSV or text as the supported formats.
+
+Default : CSV
+
+### --post-import-data
+
+Run this argument with [import schema](#import-schema) command to import indexes and triggers after data import is complete.
+The `--post-import-data` argument assumes that data import is already done and imports only indexes and triggers.
```
Comment (amit-yb) [inline docs/content/preview/migrate/yb-voyager-cli.md:343]:
```
When `--post-import-data` flag is passed, the `import schema` command imports indexes and triggers in the target DB.
```

**Review by amit-yb — COMMENTED** [review]
One minor nit.
Please address the comments provided by Rahul.
Remaining changes look okay.

File: docs/content/stable/migrate/yb-voyager-cli.md [hunk 16]
```diff
@@ -111,7 +105,29 @@ yb-voyager export data --export-dir /path/to/yb/export/dir \
         --source-db-user username \
         --source-db-password password \
         --source-db-name dbname \
-        --source-db-schema schemaName #Include schema name only for Oracle migrations.
+        --source-db-schema schemaName \ #Not applicable for MySQL.
+        --oracle-db-sid string \ #Applicable only for Oracle
```
[the remark this answers is not in the document]
Comment (aishwarya24) [reply docs/content/stable/migrate/yb-voyager-cli.md:109]: will mention this on the migration steps page. Thanks for this update.

[the remark this answers is not in the document]
Comment (aishwarya24) [reply docs/content/stable/migrate/yb-voyager-cli.md:109]: will mention this on the migration steps page. Thanks for this update.

**Comment by rahulb-yb** [comment]
LGTM, thanks for the changes!

**Pull request merged**